src/pages/Planting/PlantingLogic.tsx

```diff
@@ -29,22 +29,26 @@ export const PlantingLogic = () => {
   const history = useHistory()
 
   const walletStore = useStore($walletStore)
-  const { treesPrice } = useStore($forest)
+  const { treesInfo } = useStore($forest)
   const { selectedTreeType } = useStore($app)
   const user = useStore($user)
   const { userBalance, safeBalance } = useStore($app)
 
   const plantingTreeImages = [shihuahuacoTreeImage, cacaoTreeImage, guabaTreeImage, caobaImage]
 
   useEffect(() => {
-    setTreeImage(plantingTreeImages[treeNames.indexOf(selectedTreeType)])
-    setCurrentTreePrice(String(treesPrice[treeNames.indexOf(selectedTreeType)] * 10 ** 18))
-  }, [selectedTreeType, treesPrice])
+    if (treesInfo.length > 0) {
+      setTreeImage(plantingTreeImages[treeNames.indexOf(selectedTreeType)])
+      setCurrentTreePrice(String(treesInfo[treeNames.indexOf(selectedTreeType)].price * 10 ** 18))
```

Comment:
You can hide 10 ** 18 inside a method to avoid writing this multiplier every time you want to call a method.